mithril-aggregator/src/dependency_injection/builder.rs

```diff
@@ -502,39 +501,18 @@ impl DependenciesBuilder {
         Ok(self.multi_signer.as_ref().cloned().unwrap())
     }
 
-    async fn build_certificate_pending_store(&mut self) -> Result<Arc<CertificatePendingStore>> {
-        let adapter: Box<dyn StoreAdapter<Key = String, Record = CertificatePending>> = match self
-            .configuration
-            .environment
-        {
-            ExecutionEnvironment::Production => {
-                let adapter =
-                    SQLiteAdapter::new("pending_certificate", self.get_sqlite_connection().await?)
-                        .map_err(|e| DependenciesBuilderError::Initialization {
-                            message: "Cannot create SQLite adapter for PendingCertificate Store."
-                                .to_string(),
-                            error: Some(e.into()),
-                        })?;
-
-                Box::new(adapter)
-            }
-            _ => {
-                let adapter = MemoryAdapter::new(None).map_err(|e| {
-                    DependenciesBuilderError::Initialization {
-                        message: "Cannot create Memory adapter for PendingCertificate Store."
-                            .to_string(),
-                        error: Some(e.into()),
-                    }
-                })?;
-                Box::new(adapter)
-            }
-        };
-
-        Ok(Arc::new(CertificatePendingStore::new(adapter)))
+    async fn build_certificate_pending_store(
+        &mut self,
+    ) -> Result<Arc<dyn CertificatePendingStorer>> {
+        Ok(Arc::new(CertificatePendingRepository::new(
+            self.get_sqlite_connection().await?,
+        )))
     }
 
-    /// Get a configured [CertificatePendingStore].
-    pub async fn get_certificate_pending_store(&mut self) -> Result<Arc<CertificatePendingStore>> {
+    /// Get a configured [CertificatePendingStorer].
+    pub async fn get_certificate_pending_store(
```

Comment:
Should we rename the function as well?

```suggestion
    pub async fn get_certificate_pending_storer(
```

Reply:
I also rename `build_certificate_pending_store` to `build_certificate_pending_storer`